examples/gno.land/r/demo/boards2/board.gno

```diff
@@ -92,16 +100,31 @@ func (board *Board) DeleteThread(pid PostID) {
 	}
 }
 
-func (board *Board) Render() string {
-	s := "\\[" + newLink("post", board.GetPostFormURL()) + "]\n\n"
-	if board.threads.Size() > 0 {
-		board.threads.Iterate("", "", func(_ string, v interface{}) bool {
-			s += "----------------------------------------\n"
-			s += v.(*Post).RenderSummary() + "\n"
-			return false
-		})
+func (board *Board) Render(p Pagination) string {
+	sb := new(strings.Builder)
+	sb.WriteString("\\[")
+	sb.WriteString(newLink("post", board.GetPostFormURL()))
+	sb.WriteString("]\n\n")
+	if board.threads.Size() == 0 {
+		sb.WriteString("*No posts*")
+		return sb.String()
+	}
+
+	pg := pager.NewPager(&board.threads, p.PageSize)
+	page := pg.GetPage(p.PageNumber)
```

Comment:
Is it not possible to use something like the following?

```suggestion
	page := pager.NewPager(&board.threads, defaultPageSize).MustGetPageByPath(path)
```

Where `defaultPageSize` is a global constant and path is the render path.
This would allow to remove the `Pagination` struct.

Reply:
@jeronimoalbi the idea here is to have separation of concerns, so `render.gno` which acts as a controller will be responsible for getting pagination query string information.